promitor-agent-resource-discovery/templates/deployment.yaml

```
@@ -28,6 +28,13 @@ spec:
       {{- if .Values.rbac.create }}
       serviceAccountName: {{ template "promitor-agent-resource-discovery.serviceaccountname" . }}
       {{- end }}
+      nodeSelector:
```

We are missing an if here:
```suggestion
      {{- if .Values.nodeSelector }}
      nodeSelector:
```